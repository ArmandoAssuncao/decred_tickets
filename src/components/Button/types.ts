export interface IProps {
    m: string
    onPress: () => void
    title: string
}

export interface IState {
}
