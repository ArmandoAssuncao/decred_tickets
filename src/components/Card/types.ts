import { ITicket } from 'store/ticket/types'

export interface IProps {
    onPress?: () => void
    type: ITicket['status']
    value: number
}

export interface IState {
}
