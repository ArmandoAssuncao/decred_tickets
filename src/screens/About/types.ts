import { IScreenProps } from 'containers/Root/types'

export interface IConnectedProps {
}

export interface IActionCreators {
}

export interface IProps extends IScreenProps, IConnectedProps, IActionCreators {
}

export interface IState {
}
