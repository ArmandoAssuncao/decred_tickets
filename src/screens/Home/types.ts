import { IScreenProps } from 'containers/Root/types'
import { getTicket } from 'store/ticket/actions'
import { ITicketState } from 'store/ticket/types'

export interface IConnectedProps {
    ticket?: ITicketState
}

export interface IActionCreators {
    getTicket: typeof getTicket
}

export interface IProps extends IScreenProps, IConnectedProps, IActionCreators {
}

export interface IState {
    stakePool: ITicketState['stakePool']
    ticketAddress: ITicketState['address']
}
