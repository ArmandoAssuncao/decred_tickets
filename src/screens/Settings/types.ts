import { IScreenProps } from 'containers/Root/types'
import { permitNotifications } from 'store/settings/actions'
import { ISettingsState } from 'store/settings/types'

export interface IConnectedProps {
    settings?: ISettingsState
}

export interface IActionCreators {
    permitNotifications: typeof permitNotifications
}

export interface IProps extends IScreenProps, IConnectedProps, IActionCreators {
}

export interface IState {
    isNotifications: boolean
}
