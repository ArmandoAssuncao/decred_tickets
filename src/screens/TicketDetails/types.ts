import { IScreenProps } from 'containers/Root/types'
import { ITicket } from 'store/ticket/types'

export interface IConnectedProps {
}

export interface IActionCreators {
}

export interface IProps extends IScreenProps<INavigationState>, IConnectedProps, IActionCreators {
}

export interface IState {
}

export interface INavigationState {
    params: {
        tickets: ITicket[]
    }
}
