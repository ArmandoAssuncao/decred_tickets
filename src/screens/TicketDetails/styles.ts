import { ViewStyle } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({
    $borderRadius: 10,
    container: {
        backgroundColor: '$primaryColor',
        flex: 1,
    } as ViewStyle,
    scrollContentContainer: {
        padding: '0.4rem',
    } as ViewStyle,
    notFoundContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    } as ViewStyle,
    notFound: {
        color: '$textColorSecond',
    },
})
