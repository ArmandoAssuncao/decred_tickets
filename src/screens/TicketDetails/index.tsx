import React from 'react'
import { ScrollView, View } from 'react-native'
import { NavigationScreenConfig } from 'react-navigation'
import { connect, MapDispatchToProps } from 'react-redux'
import { bindActionCreators } from 'redux'
import { IAppState } from 'store/types'

import i18n from 'config/i18n'

import { IActionCreators, IConnectedProps, IProps, IState, } from './types'
import styles from './styles'

import { CardTicketDetail, Text } from 'components'

const mapStateToProps = (_state: IAppState): IConnectedProps =>
  ({})

const mapDispatchToProps = (dispatch: any): MapDispatchToProps<IActionCreators, IProps> =>
    bindActionCreators(
        {},
        dispatch
    )

export class TicketDetails extends React.PureComponent<IProps, IState> {
    static navigationOptions: NavigationScreenConfig<any> = {
        title: i18n.t('screens.ticketDetails.title'),
    }

    renderCardTickets = (): JSX.Element | JSX.Element[] => {
        const { tickets } = this.props.navigation.state.params
        return Array.isArray(tickets) && tickets.length
            ? tickets.map((ticket, i) => <CardTicketDetail ticket={ticket} key={i.toString()} />)
            : (
                <View style={styles.notFoundContainer}>
                    <Text
                        size="xl"
                        m="No tickets found"
                        id={'screens.ticketDetails.noTicketsFound'}
                        style={styles.notFound}
                    />
                </View>
            )
    }

    render(): JSX.Element {
        return (
            <ScrollView style={styles.container} contentContainerStyle={styles.scrollContentContainer}>
                {this.renderCardTickets()}
            </ScrollView>
        )
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(TicketDetails)
