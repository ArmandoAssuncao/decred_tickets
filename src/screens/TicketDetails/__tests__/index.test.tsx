import * as React from 'react'
import { shallowStore } from 'test/enzyme'

import TicketDetails from '../index'
import { INavigationState } from '../types'

describe('TicketDetails test render', () => {
    const store = {}
    const props = {
        navigation: {
            dispatch: jest.fn(),
            goBack: jest.fn(),
            dismiss: jest.fn(),
            navigate: jest.fn(),
            state: {
                params: {
                    tickets: [],
                },
            } as INavigationState,
        } as any,
    }
    let wrapper = shallowStore(<TicketDetails {...props} />, store)

    it('Testing snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('Testing instance of component', () => {
        expect(wrapper.instance()).toBeInstanceOf(TicketDetails.WrappedComponent)
    })

    it('has 1 ScrollView', () => {
        expect(wrapper.find('ScrollView')).toHaveLength(1)
    })

    it('has 1 Component when tickets is empty', () => {
        expect(wrapper.find('Component')).toHaveLength(1)
    })

    it('has 1 Text when tickets is empty', () => {
        expect(wrapper.find('Text')).toHaveLength(1)
    })

    it('has 1 CardTicketDetail when tickets length is 1', () => {
        const newProps = { ...props }
        newProps.navigation.state = {
            params: {
                tickets: [
                    {
                        buyTime: 0,
                        buyTxId: 'buyTxId',
                        feeStakePool: 0,
                        returnTime: 0,
                        returnTotal: 0,
                        returnTxId: 'returnTxId',
                        reward: 0,
                        status: 'live',
                        ticketFee: 0,
                        ticketPrice: 0,
                        totalInvestment: 0,
                    },
                ],
            },
        } as INavigationState
        wrapper = shallowStore(<TicketDetails {...props} />, store)
        expect(wrapper.find('CardTicketDetail')).toHaveLength(1)
    })
})
