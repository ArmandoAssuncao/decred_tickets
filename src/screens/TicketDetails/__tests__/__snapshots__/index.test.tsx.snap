// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TicketDetails test render Testing snapshot 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "padding": 6.4,
    }
  }
  style={
    Object {
      "backgroundColor": "#0C1E3E",
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      id="screens.ticketDetails.noTicketsFound"
      m="No tickets found"
      size="xl"
      style={
        Object {
          "color": "#FFF",
        }
      }
      type="normal"
    />
  </Component>
</ScrollView>
`;
